ddtrace/contrib/aiohttp/middlewares.py

```diff
@@ -47,8 +47,9 @@ def attach_context(request):
         )
 
         # Configure trace search sample rate
-        if config.aiohttp.event_sample_rate is not None:
-            request_span.set_tag(EVENT_SAMPLE_RATE_KEY, config.aiohttp.event_sample_rate)
+        analytics_sample_rate = config.aiohttp.get_analytics_sample_rate()
+        if analytics_sample_rate:
```

Comment:
What if the same rate is 0?

Reply:
If rate is 0, then the metric is not set, so the effect would be the same as disabling analytics.

While the end result of setting it to 0 should be to not have traces consumed by Trace Search, I can see squashing 0 and `None` in this case as bad practice for code clarity and debugging.

Reply:
The result is not the same if it's also configured in the agent.

Reply:
Correct, we need `0` and "not set" to be different values.

If they want to explicitly set as `0` to disable analytics for a given integration then what you have here will not set it meaning it'll use whatever is set in the agent, which may not be `0`.

If that makes sense?